### DISCUSSION TITLE
Cannot Push Local Branch for Pull Request - Need Assistance

### DISCUSSION THREAD

**Discussion by andreshat** [nlmatics/llmsherpa#66] [Q&A]
Hello, wonderful community,

I've encountered an issue while trying to contribute to the project. Despite making several enhancements and fixes locally, including improving HTML table rendering, fixing `ul` tag usage, and refining code styling, I am unable to push my local branch to prepare a pull request. 

Here's a brief overview of the changes I've made:
- Fixed HTML table rendering issues (ensured `th` tags are properly contained within `tr` tags).
- Corrected the misuse of `ul` tags outside of `p` tags.
- Improved table rendering to remove unnecessary spacing before table rows and ensured proper closure with vertical lines.
- Added better code styling for readability, including necessary newlines.

Unfortunately, when attempting to push my branch, I receive an error (or am simply unable to push, depending on the specific circumstances - kindly adjust based on the exact issue you're encountering).

Has anyone else faced this issue, or does anyone have insights on what might be going wrong? I'm eager to contribute these improvements to our project and would greatly appreciate any guidance or steps I might follow to resolve this.

Thank you in advance for your help and support!

Best,
Andrii Shatokhin

**Comment by ansukla** [comment]
Hi @andreshat  - Thank you for making the contributions. Please submit a PR with your changes - once the PR is reviewed, it can be merged. This is the process for most open source/closed source codebases.

**Reply by andreshat** [reply]
The issue is with granted access. I have no option to Push to the project. At first I have cloned repo with url, but for now I have updated origin with ssh and still have problems with pushing. Any ideas how to overcome this?
<img width="518" alt="image" src="https://github.com/nlmatics/llmsherpa/assets/28289695/e20257fc-a74d-4046-add4-87d232bd2f48">

Push failed ERROR: Permission to nlmatics/llmsherpa.git denied to andreshat. Could not read from remote repository. Please make sure you have the correct access rights and the repository exists.

**Comment by ansukla** [comment]
Hi @andreshat - I think you will first have to fork this repo to your handle and then make the changes and push. The process is described here: https://github.com/firstcontributions/first-contributions. Let me know if you followed this process. There have been quite a few PRs and I don't think this is an access problem. Thanks again for your contribution!
